cryptomatte/cryptomatte.h

```diff
@@ -1072,9 +1072,9 @@ struct CryptomatteData {
                 crypto_aovs_count++;
                 setup_new_outputs(universe, t_output, crypto_aovs, outputs_new);
 
-                if (AiNodeEntryLookUpParameter(AiNodeGetNodeEntry(driver), "half_precision")) {
-                    if (AiNodeGetBool(driver, "half_precision")) {
-                        AiNodeSetBool(driver, "half_precision", false);
+                if (AiNodeEntryLookUpParameter(AiNodeGetNodeEntry(driver), AtString("half_precision"))) {
+                    if (AiNodeGetBool(driver, AtString("half_precision"))) {
+                        AiNodeSetBool(driver, AtString("half_precision"), false);
```

Comment:
Creating an `AtString` is expensive as a hash has to be computed and there's some critical sections where the underlying `AtString` data is being saved into some structures. It's recommended to compute these once and then reuse them. One other advantage of this is that it makes it less likely that typos appear (how easy would it be to detect that one of you 10 usages of `AtString("half_precision)" doesn't have a typo? I suspect that kind of bug would be hard to track down).

In this particular case, it's for `setup_outputs` which doesn't seem performance critical (happens only once I imagine), but it's good practice when possible to do this optimization everywhere.

What I like is to create a file that contains all the `AtString` I might need. Something like:
```c++
namespace ATSTR
{
// use REGISTER_STRING when registering a string that is a valid variable name
//
// use REGISTER_STRING2 when the string is not a valid variable name and so
// won't compile.  For instance, because it is made up of multiple words:
// AtString my variable("my variable"); XXX won't compile
// or because that name is already taken:
// AtString NULL("NULL"); XXX NULL is already defined to be 0 or __null
// AtString if("if"); XXX if is a keyword used for if statements

#ifdef ATSTRINGS_CPP
// pretend this file is an "atstrings.cpp" where we need the definitions. Only call
// this from one cpp file
#define REGISTER_STRING(x)       extern const AtString x; const AtString x(#x)
#define REGISTER_STRING2(x, str) extern const AtString x; const AtString x(str)
#else
// included from other source files, where we only need the declarations
#define REGISTER_STRING(x)       extern const AtString x
#define REGISTER_STRING2(x, str) extern const AtString x
#endif

   REGISTER_STRING(half_precision);
   REGISTER_STRING(outputs);
...
```
Then you would use it like this:
```suggestion
                if (AiNodeEntryLookUpParameter(AiNodeGetNodeEntry(driver), ATSTR::half_precision)) {
                    if (AiNodeGetBool(driver, ATSTR::half_precision)) {
                        AiNodeSetBool(driver, ATSTR::half_precision, false);
```